libs/acn/PDU.cpp

```diff
@@ -117,13 +123,18 @@ bool PDU::Pack(uint8_t *buffer, unsigned int *length) const {
 void PDU::Write(OutputStream *stream) const {
   unsigned int size = Size();
 
-  if (size <= TWOB_LENGTH_LIMIT) {
+  if (size <= TWOB_LENGTH_LIMIT && !m_force_length_flag) {
     uint16_t flags_and_length = static_cast<uint16_t>(size);
     flags_and_length |= (VFLAG_MASK | HFLAG_MASK | DFLAG_MASK) << 8u;
     *stream << HostToNetwork(flags_and_length);
   } else {
     uint8_t vhl_flags = static_cast<uint8_t>((size & 0x0f0000) >> 16);
     vhl_flags |= VFLAG_MASK | HFLAG_MASK | DFLAG_MASK;
+    if (m_force_length_flag) {
+      // TODO(Peter): Should this happen regardless of the force as we're
```

Comment:
same here

Reply:
Likewise.

Reply:
I've added a note to check these in #1841